### DISCUSSION TITLE
msegui & bgrabitmap

### DISCUSSION THREAD

**Discussion by codz01** [mse-org/mseide-msegui#55] [General]
Hi
is bgrabitmap compatible with msegui ?

**Comment by rchastain** [comment]
Hi. Yes. For example, this project uses BGRABitmap: [strumpract](https://github.com/fredvs/strumpract)
(Look at the last screenshot, at the end of the page.)

**Comment by fredvs** [comment]
Yes, of course, since version [5.2.0.].

There is even a demo in BGRABitmap source with **fractal_tree.prj** mseide/ideU project file:
https://github.com/bgrabitmap/bgrabitmap/tree/dev-bgrabitmap/test/test4ideu/fractal_tree

![bgra_mseguitext](https://user-images.githubusercontent.com/3421249/173422806-deabce4f-0198-464f-8279-7c0805c0a05f.png)

[EDIT] You will need to create the directory **/units** in the root directory of the project (empty directories are removed by GitHub)

**Comment by codz01** [comment]
thanks , i download bgrabitmap and tryed open the project you mention  ( fractal_tree ) , but i got lots of errors , do i need extra steps to compile this project ?
![Untitled](https://user-images.githubusercontent.com/10813427/173449759-3c3367d8-2888-4a7f-ad97-3425969b5e39.jpg)

**Comment by codz01** [comment]
i am using msegui 5.6.7
![about](https://user-images.githubusercontent.com/10813427/173450059-2ed8ab54-e98f-4fc5-9350-2023e487b391.jpg)

**Comment by fredvs** [comment]
Hello.

Did you set in _Project Options-Make-Directories_ the dev-directory of BGRABitmap?

![fractral_tree_dir](https://user-images.githubusercontent.com/3421249/173457313-cf6284ee-937a-42d0-aeb6-acd22cf277e1.png)

You will need to download the source of BGRABitmap from here:
https://github.com/bgrabitmap/bgrabitmap/archive/refs/heads/dev-bgrabitmap.zip

Unzip it and set the directory of /dev-bgrabitmap/bgrabitmap in project options-make-directories.

To use BGRABitmap with MSEgui, you will need those fpc parameters:
**_-dclass_bridge_** and **_-dBGRABITMAP_USE_MSEGUI_**

![fractral_tree_par](https://user-images.githubusercontent.com/3421249/173574344-04b6dc98-663f-40a2-906b-ac3705e5d58f.png)


And re-build all (better to first delete all the files in /units).

**Comment by codz01** [comment]
Hi
sorry for the delay , 
it was my mistake i forgot the -B switch , msegui should rebuild with class_bridge define 
it works fine now , thanks for the info

**Comment by fredvs** [comment]
>  i forgot the -B switch 

Aaargh, that "eternal" fpc bug is still not fixed, indeed, without **-B**, fpc ignore the content inside some {$ ifdef ...}.
